src/applications/app.helpers.ts

```diff
@@ -50,6 +57,38 @@ export function isParcel(appOrParcel) {
   return Boolean(appOrParcel.unmountThisParcel);
 }
 
-export function objectType(appOrParcel) {
+export function objectType(appOrParcel: AppOrParcel): "parcel" | "application" {
   return isParcel(appOrParcel) ? "parcel" : "application";
 }
+
+export interface InternalApplication {
```

Comment:
This replaces the `ApplicationObject` type that was incomplete within lifecycle.helpers.ts